app/file-detail/controller.js

```diff
@@ -1,25 +1,105 @@
 import Controller from '@ember/controller';
 
 export default Controller.extend({
+    currentUser: service(),
+    revision: null,
+    edit: false,
     displays: A([]),
-    userId: computed('model', function() {
-        // let link = this.get('model')._internalModel._relationships.initializedRelationships.user.link; //hack to get id without having to request it
-        // return link.split('/')[link.split('/').indexOf('users') + 1];
-        const model = this.get('model');
-        return model.get('user').then(user => user.get('id'));
+
+    mfrVersion: computed('model.file', 'revision', function() {
+        return this.get('revision') ? this.get('revision') : this.get('model.file.currentVersion');
+    }),
+
+    fileUrl: computed('model.file', function() {
+        return encodeURIComponent(window.location.href);
+    }),
+
+    twitterUrl: computed('model.file', 'fileUrl', function() {
+        return `https://twitter.com/intent/tweet?url=${this.get('fileUrl')}&text=${this.get('model.file.name')}&via=OSFramework`;
+    }),
+
+    facebookUrl: computed('model.file', 'fileUrl', function() {
+        return `https://www.facebook.com/sharer/sharer.php?u=${this.get('fileUrl')}`;
+    }),
+
+    linkedInUrl: computed('model.file', 'fileUrl', function() {
+        return `https://www.linkedin.com/cws/share?url=${this.get('fileUrl')}&title=${this.get('model.file.name')}`;
     }),
+
+    emailUrl: computed('model.file', 'fileUrl', function() {
+        return `mailto:?subject=${this.get('model.file.name')}&body=${this.get('fileUrl')}`;
+    }),
+
+    mfrUrl: computed('model.file', function() {
+        return `https://mfr.osf.io/render?url=${window.location.href}?action=download%26mode=render`;
+    }),
+
+    shareiFrameDynamic: computed('model.file', function() {
+        return `<style>.embed-responsive{position:relative;height:100%;}.embed-responsive iframe{position:absolute;height:100%;}</style><script>window.jQuery || document.write('<script src="//code.jquery.com/jquery-1.11.2.min.js">\x3C/script>') </script><link href="https://mfr.osf.io/static/css/mfr.css" media="all" rel="stylesheet"><div id="mfrIframe" class="mfr mfr-file"></div><script src="https://mfr.osf.io/static/js/mfr.js"></script> <script>var mfrRender = new mfr.Render("mfrIframe", "${this.get('mfrUrl')}");</script>`;
+    }),
+
+    shareiFrameDirect: computed('model.file', function() {
+        return `<iframe src="${this.get('mfrUrl')}" width="100%" scrolling="yes" height="677px" marginheight="0" frameborder="0" allowfullscreen webkitallowfullscreen>`;
+    }),
+
+    fileTags: computed('model.file', function() {
+        return this.get('model.file.tags');
+    }),
+
+    fileVersions: computed('model.file', function() {
+        return $.getJSON(`${this.get('model.file.links.download')}?revisions=&`).then(function(data) {
+            return data.data;
+        });
+    }),
+
+    _edit: observer('currentUser', 'model.user', function() {
```

Comment:
I think there's a way to do this without an observer. You could make `edit` a computed property that returns the value of `_edit`, which would be initialized to `false`. In the computed property, if `model.user.id` is loaded, just update `_edit` before you return it.